### PR TITLE
BR-42 Upgrade to maven-latest

### PR DESCRIPTION
**Pull request by TonyMorrison**
### Requirements
* Filling out the template is required. Any pull request that does not
include enough information to be reviewed in a timely manner may be
closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change
The maven version used in the pipeline has been changed from 3.5.x to maven-latest, which currently is 3.6.3.

### Benefits
This change ensures we are building with the most current version of Maven

### Verification Process
The changes have been applied and tested from my fork of the code
http://jenkins.phx.connexta.com/service/jenkins/job/W.I.P./job/codice-testTM/job/master/

### CODE REVIEW

**Comment by TonyMorrison** [comment]
build now

File: Jenkinsfile [hunk 1]
```diff
@@ -249,7 +249,7 @@ pipeline {
                     }
                 }
 
-                withMaven(maven: 'Maven 3.5.3', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
+                withMaven(maven: 'maven-latest', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
```
Comment (LinkMJB) [inline Jenkinsfile]: There might be negative impacts during the deploy stage (i.e. if maven-deploy-plugin needs to be pinned/upgraded). I'm not sure we will really know, until we merge this PR though.

File: Jenkinsfile [hunk 2]
```diff
@@ -29,7 +29,7 @@ pipeline {
           Restrict nightly builds to master branch, all others will be built on change only.
           Note: The BRANCH_NAME will only work with a multi-branch job using the github-branch-source
         */
-        cron(BRANCH_NAME == "master" ? "H H(17-19) * * *" : "")
+        pollSCM(BRANCH_NAME == "master" ? "H H(17-19) * * *" : "")
```
Comment (LinkMJB) [inline Jenkinsfile]: Any risk with this not being built frequently, if changes are infrequent? I know with builds we only were doing pollSCM on in the past, there were concerns that "CI was not running for the project."

Comment (TonyMorrison) [reply Jenkinsfile]: i could change it to cron/weekly if that makes better sense. will it still mark it as no CI running if it runs weekly?

Comment (TonyMorrison) [reply Jenkinsfile]: changed back to nightly. We should have the discussion regarding the frequency of building apps that are rarely modified.

**Comment by TonyMorrison** [comment]
build now

**Pull request merged**